### DISCUSSION TITLE
Using DelayedLazyView with appShell

### DISCUSSION THREAD

**Discussion by fgiacomelli** [roubachof/Sharpnado.Tabs#65] [General]
Hello, I'm writing an app that will have a bottom navigation tab and complex views inside each tab.
I want to use native appshell as tabs, but I would like to use the delayedview because my views requires too much time to be rendered to give to the user a good experience.
Could it be possible that I just use the delayedview to load my pages without use the tabhostview?

**Comment by roubachof** [comment]
Of course! Just wrap your page root layout in a delayed view and call load() in onAppearing in the code behind

**Reply by fgiacomelli** [reply]
> Of course! Just wrap your page root layout in a delayed view and call load() in onAppearing in the code behind

Thank you for your reply, but I should have missed some configurations...if I wrap my layout in a delayed view, like this:

` <sharpnado:DelayedView x:Name="delayedView" UseActivityIndicator="True" >
        <Grid Margin="0,0,0,0">`

and then load the view in OnAppearing:

`    protected override void OnAppearing()
    {
        base.OnAppearing();
        delayedView.LoadView();
    }
`

I can only see a totally white empty page. I registered sharpnado inside the CreateMauiApp method in MauiProgram file with

`  .UseSharpnadoTabs(loggerEnable: false)`

**Reply by roubachof** [reply]
Sorry I said some total nonsense :)
You have to give a type to the DelayedView.
So it means that you have to put the child into a separate class.
For example your child seem to be a Grid, so you have to create a custom class extending grid, lets name it MyChild, and put your xaml in it.
Then:

```xml
<tabs:DelayedView x:TypeArguments="views:MyChild"
                              AccentColor="{StaticResource Primary}"
                              Animate="True"
                              BindingContext="{Binding MyPageViewModel}"
                              UseActivityIndicator="True" />
```
